test/test_transforms.py

```diff
@@ -1200,6 +955,292 @@ def test_randomness(fn, trans, config, p):
+    yield img_data_byte, expected_output, 'L'
+
+    img_data_short = torch.ShortTensor(1, 4, 4).random_()
+    expected_output = img_data_short.numpy()
+    yield img_data_short, expected_output, 'I;16'
+
+    img_data_int = torch.IntTensor(1, 4, 4).random_()
+    expected_output = img_data_int.numpy()
+    yield img_data_int, expected_output, 'I'
+
+
+@pytest.mark.parametrize('pass_mode', [False, True])
+@pytest.mark.parametrize('img_data, expected_output, expected_mode', _get_1_channel_tensor_various_types())
+def test_1_channel_tensor_to_pil_image(pass_mode, img_data, expected_output, expected_mode):
+    transform = transforms.ToPILImage(mode=expected_mode) if pass_mode else transforms.ToPILImage()
+    to_tensor = transforms.ToTensor()
+
+    img = transform(img_data)
+    assert img.mode == expected_mode
+    torch.testing.assert_close(expected_output, to_tensor(img).numpy(), check_stride=False)
+
+
+def test_1_channel_float_tensor_to_pil_image():
+    img_data = torch.Tensor(1, 4, 4).uniform_()
+    # 'F' mode for torch.FloatTensor
+    img_F_mode = transforms.ToPILImage(mode='F')(img_data)
+    assert img_F_mode.mode == 'F'
+    torch.testing.assert_close(
+        np.array(Image.fromarray(img_data.squeeze(0).numpy(), mode='F')), np.array(img_F_mode)
+    )
+
+
+@pytest.mark.parametrize('img_data, expected_mode', [
+    (torch.Tensor(4, 4, 1).uniform_().numpy(), 'F'),
+    (torch.ByteTensor(4, 4, 1).random_(0, 255).numpy(), 'L'),
+    (torch.ShortTensor(4, 4, 1).random_().numpy(), 'I;16'),
+    (torch.IntTensor(4, 4, 1).random_().numpy(), 'I'),
+])
+def test_1_channel_ndarray_to_pil_image(img_data, expected_mode):
+    for transform in [transforms.ToPILImage(), transforms.ToPILImage(mode=expected_mode)]:
+        img = transform(img_data)
+        assert img.mode == expected_mode
+        # note: we explicitly convert img's dtype because pytorch doesn't support uint16
+        # and otherwise assert_close wouldn't be able to construct a tensor from the uint16 array
+        torch.testing.assert_close(img_data[:, :, 0], np.asarray(img).astype(img_data.dtype))
+
+
+@pytest.mark.parametrize('expected_mode', [None, 'LA'])
+def test_2_channel_ndarray_to_pil_image(expected_mode):
+    img_data = torch.ByteTensor(4, 4, 2).random_(0, 255).numpy()
+
+    if expected_mode is None:
+        img = transforms.ToPILImage()(img_data)
+        assert img.mode == 'LA'  # default should assume LA
+    else:
+        img = transforms.ToPILImage(mode=expected_mode)(img_data)
+        assert img.mode == expected_mode
+    split = img.split()
+    for i in range(2):
+        torch.testing.assert_close(img_data[:, :, i], np.asarray(split[i]), check_stride=False)
+
+
+def test_2_channel_ndarray_to_pil_image_error():
+    img_data = torch.ByteTensor(4, 4, 2).random_(0, 255).numpy()
+    transforms.ToPILImage().__repr__()
+
+    with pytest.raises(ValueError, match=r"Only modes \['LA'\] are supported for 2D inputs"):
```

Comment:
The original code was wrong here, but we'll need one `raises` block per line, because we want to make sure that each individual line effectively raises an error

Reply:
Done as suggested!